test/aws-config/mzcompose.py

```diff
@@ -0,0 +1,320 @@
+    volumes_extra=AWS_VOLUME,
+)
+
+
+def workflow_default(c: Composition) -> None:
+    "Test that materialize can use a multitude of auth schemes to connect to AWS"
+    LOCAL_DIR.mkdir()
+
+    session = boto3.Session()
+    sts: STSClient = session.client("sts")
+    iam: IAMClient = session.client("iam")
+
+    identity = sts.get_caller_identity()
+    current_user = identity["Arn"]
+
+    aws_region = session.region_name
+
+    created_roles: List[CreatedRole] = []
+    try:
+        allowed = create_role(iam, "Allow", current_user, created_roles)
+        denied = create_role(iam, "Deny", current_user, created_roles)
+        requires_eid = create_role(
+            iam, "Allow", current_user, created_roles, external_id=EXTERNAL_ID
+        )
+        profile_contents = gen_profile_text(
+            session, allowed.arn, requires_eid.arn, denied.arn
+        )
+
+        wait_for_role(sts, allowed.arn)
+
+        td_args = [
+            f"--aws-region={aws_region}",
+            f"--var=allowed-role-arn={allowed.arn}",
+            f"--var=denied-role-arn={denied.arn}",
+            f"--var=role-requires-eid={requires_eid.arn}",
+        ]
+
+        # == Run core tests ==
+
+        c.up("materialized")
+
+        write_aws_config(LOCAL_DIR, profile_contents)
+
+        c.wait_for_materialized("materialized")
+        c.run(
+            "testdrive-svc",
+            *td_args,
+            "test.td",
+        )
+        c.run(
+            "testdrive-svc",
+            *td_args,
+            # no reset because the next test wants to validate behavior with
+            # the previous catalog
+            "--no-reset",
+            "test-externalid-missing.td",
+        )
+
+        # == Tests that restarting materialized without a profile doesn't bork mz ==
+
+        print("+++ Test Restarts with and without profile files")
+
+        # Historically, a missing aws config file would cause all SQL
+        # commands to hang entirely after a restart, this no longer happens
+        # but this step restarts to catch it if it comes back.
+        c.stop("materialized")
+
+        rm_aws_config(LOCAL_DIR)
+
+        c.up("materialized")
+
+        c.run(
+            "testdrive-svc",
+            "--no-reset",
+            "test-restart-no-creds.td",
+        )
+
+        # now test that with added credentials things can be done
+        write_aws_config(LOCAL_DIR, profile_contents)
+        c.run("testdrive-svc", *td_args, "test-restart-with-creds.td")
+
+        # == Test that requires --aws-external-id has been supplied ==
+        print("+++ Test AWS External IDs")
+        c.stop("materialized")
+        c.rm("materialized")
+
+        with c.override(MZ_EID):
+            c.up("materialized")
+            c.wait_for_materialized("materialized")
+            write_aws_config(LOCAL_DIR, profile_contents)
+            c.run("testdrive-svc", *td_args, "test-externalid-present.td")
```

Comment:
Just musing, and 100% not the fault of this PR or something to address here, but using testdrive for tests like this splits the logic of the test across Python and testdrive scripts in a way that's hard to follow. I'm hoping to add the barest of SQL assertions to mzcompose soon, so that simple tests like this can be written all in Python.